src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/PrefetchGetRecordsCache.java

```diff
@@ -138,6 +151,7 @@ public void run() {
                     log.warn("Prefetch thread was interrupted.");
                     break;
                 }
+                MetricsHelper.startScope(metricsFactory, operation);
```

Comment:
Will fail if operation is not set, take the operation as a parameter to the constructor

Reply:
The GetRecordsCache was created inside the constructor of ShardConsumer, I don't think the shardConsumer can get the operation's name inside its constructor.

Reply:
The getRecordsCache gets started in the InitializeTask, which comes before the ProcessTask is called. This would cause the given line to fail.

Reply:
Initialize task and ProcessTask needs different operation name, using constructor to pass the operation would still require set the operation when ProcessTask calls it. Add the setMetricOpertion call to initialize task should be able to fix it.

Reply:
The InitializeTask starts the cache. This will try to use the metrics, if the value is not set in the initialize task, it will cause the cache to keep on failing. This will lead to no records gotten from Kinesis, which in turn will lead to no calls being made to the process task.